src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -81,4 +89,29 @@ public Date deserialize(JsonElement jsonElement, Type type,
             }
         }
     }
+
+    private class StatsDeserializer implements JsonDeserializer<Stats> {
+
+        @Override
+        public Stats deserialize(JsonElement jsonElement, Type type,
+                JsonDeserializationContext jsonDeserializationContext)
+                throws JsonParseException {
+            try {
+                JsonObject stats = jsonElement.getAsJsonObject();
+                Applications app = gson.fromJson(stats.get("CherryPy Applications"), Applications.class);
+                HttpServer server = null;
+                for(Map.Entry<String, JsonElement> entry : stats.entrySet()){
+                    String key = entry.getKey();
+                    if(key.startsWith("CherryPy HTTPServer ")){
+                        server = gson.fromJson(entry.getValue(), HttpServer.class);
+                        break;
+                    }
+                }
+                return new Stats(app, server);
```

Comment:
@moio You are right that's a really good way of putting it. Since it says CherryPy HTTPServer i would guess thats is statistics about the http server we are using to make the api calls. So unless those statistics can be disabled in some config i would expect them always to be there if we get data back from the server. I will change it to throw an exception